### ISSUE TITLE
Inconststent mapping of array with one item vs array with more than one items

### ISSUE THREAD

**Issue by tolomaus** [51zero/eel-sdk#233]
See here how to reproduce the error:

In Spark 2.1.0:
```scala
case class WordVector(word: String, vector: Array[Double])

Seq(WordVector("abc", Array(1.0, 2.0, 3.0)), WordVector("abc", Array(1.0, 2.0))).toDF.write.save("/tmp/WordVectorsGood")

Seq(WordVector("abc", Array(1.0, 2.0, 3.0)), WordVector("abc", Array(1.0))).toDF.write.save("/tmp/WordVectorsBad")
```

Note there is an array with one item in the bad dataset

Using eels 1.1.1:
```scala
ParquetSource(new Path("/tmp/WordVectorsGood")).parts.flatMap(part => part.iterator.flatMap(identity).map(row => row.values(1).asInstanceOf[Seq[Seq[Seq[Double]]]].flatten.flatten.toArray).toList)

ParquetSource(new Path("/tmp/WordVectorsBad")).parts.flatMap(part => part.iterator.flatMap(identity).map(row => row.values(1).asInstanceOf[Seq[Seq[Seq[Double]]]].flatten.flatten.toArray).toList)
```

The good dataset loads correctly, the second one throws a class cast exception because the Seq[Double] is unwrapped into a Double under the hood

It's also not clear to me why the value is a triple nested structure.

**Comment by sksamuel** [comment]
Right I understand why the issue with a single value not being wrapped in an array is happening, so that I can fix immediately.

Why its a triple nested structure I don't know, that I'll investigate.

Thanks for the bug report.

Also @hannesmiller was wondering what made you choose to use eel over spark for reads.

**Comment by tolomaus** [comment]
Cool looking forward to the next version then ;-)

Why eel over spark: I need the word vectors in an online multi-user "low latency" context where the whole dataset still fits in memory. I have looked at Spark Job Server but it feels like a huge overhead compared to simply loading the parquet file with eel and storing it in memory (btw both the load of the file as well as the calculations afterward happen in parallel as I keep the data in a ParMap). As soon as the case class mapping is supported I think this would be a perfect solution for me.

**Comment by sksamuel** [comment]
Ok thanks for the feedback. Case class mapping will be coming in 1.2.

**Comment by sksamuel** [comment]
The reason for the nested arrays is because Spark creates a parquet schema for your case class like this:
```
message spark_schema {
  optional binary word (UTF8);
  optional group vector (LIST) {
    repeated group list {
      required double element;
    }
  }
}
```

I don't know why it creates the outer group for the vector and then creates a nested list inside that. Parquet supports lists directly, there's no need to create two levels, but perhaps there is a reason. Maybe @hannesmiller you know?

**Comment by sksamuel** [comment]
I wonder if its so they can distinguish between null list, and an empty list.

**Comment by sksamuel** [comment]
This is fixed in 1.2.0-M2.
